pkg/virtctl/create/vm/vm.go

```diff
@@ -1276,17 +1309,182 @@ func createDataVolume(spec *cdiv1.DataVolumeSpec, size string, name string, vm *
 	return nil
 }
 
+func (c *createVM) withAccessCredential(vm *v1.VirtualMachine) error {
+	for _, accessCred := range c.accessCreds {
+		src := &accessCredential{}
+		if err := params.Map(AccessCredFlag, accessCred, src); err != nil {
+			return err
+		}
+
+		if src.Source == "" {
+			return params.FlagErr(AccessCredFlag, "src must be specified")
+		}
+
+		namespace, name, err := params.SplitPrefixedName(src.Source)
+		if err != nil {
+			return params.FlagErr(AccessCredFlag, "src invalid: %w", err)
+		}
+		if namespace != "" {
+			return params.FlagErr(AccessCredFlag, "not allowed to specify namespace of secret \"%s\"", name)
+		}
+
+		var apiAccessCred *v1.AccessCredential
+		switch strings.ToLower(src.Type) {
+		case AccessCredTypeSSH, "":
+			if apiAccessCred, err = c.withAccessCredentialSSH(src); err != nil {
+				return err
+			}
+		case AccessCredTypePassword:
+			if apiAccessCred, err = c.withAccessCredentialPassword(src); err != nil {
+				return err
+			}
+		default:
+			return params.FlagErr(AccessCredFlag, "invalid access credential type \"%s\", supported values are: %s, %s", src.Type, AccessCredTypeSSH, AccessCredTypePassword)
+		}
+
+		vm.Spec.Template.Spec.AccessCredentials = append(vm.Spec.Template.Spec.AccessCredentials, *apiAccessCred)
+	}
+
+	return nil
+}
+
+func (c *createVM) withAccessCredentialSSH(src *accessCredential) (*v1.AccessCredential, error) {
+	apiAccessCred := &v1.AccessCredential{
+		SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
+			Source: v1.SSHPublicKeyAccessCredentialSource{
+				Secret: &v1.AccessCredentialSecretSource{
+					SecretName: src.Source,
+				},
+			},
+		},
+	}
+
+	switch strings.ToLower(src.Method) {
+	case AccessCredMethodGA, "":
+		if err := c.withAccessCredentialSSHMethodGA(src, apiAccessCred); err != nil {
```

Comment:
supernit - move the definition of `apiAccessCred` into each function and just return it each time? Reduces the line count in `withAccessCredentialSSH` and avoids the reader having to scroll back and fourth for context on `apiAccessCred`.

Reply:
ACK